lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -493,7 +493,7 @@ public function syntaxError($expected = '', $token = null)
         $message .= $expected !== '' ? sprintf('Expected %s, got ', $expected) : 'Unexpected ';
         $message .= $this->lexer->lookahead === null ? 'end of string.' : sprintf("'%s'", $token['value']);
 
-        throw QueryException::syntaxError($message, QueryException::dqlError($this->query->getDQL()));
+        throw QueryException::syntaxError($message, QueryException::dqlError($this->query->getDQL() ?? ''));
```

Comment:
When does `getDQL()` return null?

Reply:
It's `null` by default, when `setDQL()` is never called or when `free()` is called.